### PR TITLE
Clarify code review checklist instructions

### LINKED ISSUE #12054: Clarify how to fill the code Review Checklist:
It is not very clear how to fill the new Code Review Checklist.

### Steps to reproduce:

* Make a new PR for NVDA.
* In the code Review Checklist check the items that are applicable

### Actual behavior:

It is not clear when an item should be checked. Also maybe a comment to indicate how to check these check boxes is needed. We can already see in the few new PRs that people seem to use this list diversely according to situations:
* In #12051: the checkboxes have been removed since * was used to check it and the space was removed when it is not checked. Also "System tests" has been checked but no system test was implemented or modified. Is it an error or has it been checked only to indicate that this point has been considered (and even commented)?
* In #12042, only relevant items have been kept; the other items have been removed.

Should unit test be checked if the PR was unit tested or if a unit test was added? Should help context be checked only if a new GUI item was added or also if I have checked that my PR does not need to implement help context.

More generally:
* For items not applicable, should they be removed or unchecked
* Will a PR be reviewed only when all items are checked. This would mean that PRs with unchecked items should remain drafts.

### Expected behavior:

* Clarify when an item should be checked.
* Indicate if some items may be removed and in which case
* Maybe the checkboxes are not the right tool? E.g. if answers yes/no/NotApplicable are required.

#### NVDA installed/portable/running from source:
source
#### NVDA version:
Last alpha
#### Windows version:
N/A
#### Name and version of other software in use when reproducing the issue:
N/A
#### Does the issue still occur after restarting your computer?
N/A
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
N/A
#### If addons are disabled, is your problem still occuring?
N/A
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
N/A

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
Fixes #12054

### Summary of the issue:
Instructions for the how to use and the intentions of the code review checklist were not clear enough.

### Description of how this pull request fixes the issue:
Clarifies the instructions.
The intent of the checklist is to serve purely as a reminder, checking items communicates that you have considered the item, and have not accidentally forgotten about it.
Further, the checklist acts as a reminder for reviewers, unfortunately GitHub does not yet provide a way to create a dedicated code review template.

Additionally the wiki page has been adjusted: https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples

### Testing strategy:
This PR serves as an example.

### Known issues with pull request:
None

### Change log entry:
N/A

### Code Review Checklist:

This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review and confirm you have considered the following items.
Mark items you have considered by checking them.
You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
You can do also check the checkboxes after the PR is created.

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

### CODE REVIEW

**Comment by feerrenrut** [comment]
@CyrilleB79 please take a look at this change.

**Comment by Bung2528** [comment]
ดาวน์โหลด Outlook for iOS<https://aka.ms/o0ukef>
________________________________
จาก: Reef Turner <notifications@github.com>
ส่ง: Wednesday, February 10, 2021 11:01:29 AM
ถึง: nvaccess/nvda <nvda@noreply.github.com>
สำเนาถึง: Subscribed <subscribed@noreply.github.com>
ชื่อเรื่อง: Re: [nvaccess/nvda] Clarify code review checklist instructions (#12056)


@CyrilleB79<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FCyrilleB79&data=04%7C01%7C%7Ca4951017075143b0b84808d8cd788c0c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637485264911547226%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=mufcZOyVz5Vpllyta%2BL0PCbUk5gsh31t%2FPS9qoqUYBw%3D&reserved=0> please take a look at this change.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -20,8 +20,10 @@ Section: New features, Changes, Bug fixes
 ### Code Review Checklist:
 
 This checklist is a reminder of things commonly forgotten in a new PR.
-Please do a self-review to check these items.
-Reviewers will not approve the PR until these are met.
+Authors, please do a self-review and confirm you have considered the following items.
+Mark items you have considered by checking them.
+You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
+You can do also check the checkboxes after the PR is created.
```
Comment (CyrilleB79) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: This sentence sounds strange to me. Is there any extra "do" in it?
Feel free to ignore my suggestion if the original is correct English (using "do" to insist), I am not a native English speaker.

```suggestion
You can also check the checkboxes after the PR is created.
```

**Review by CyrilleB79 — DISMISSED** [review]
Thank you Reef. Your intent has been clarified.
Let's then see how it is used by PR authors and reviewers.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (feerrenrut) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: Yes you are right. Good catch.

**Comment by XLTechie** [comment]
Should that say:

can do also check the checkboxes after the PR is created

Or should it say:

can also check the checkboxes after the PR is created

**Comment by feerrenrut** [comment]
Oops. Yes, small typo there. Happy to accept a PR to correct!

**Pull request merged**